enhancements/vsphere-multi-vcenter.md

```diff
@@ -0,0 +1,839 @@
+	Config       *vsphere.Config
+	LegacyConfig *legacy.VSphereConfig
+}
+```
+
+The operator has also been enhanced in all of the checks that are performed.  
+Currently each check assumed everything was against one vCenter.  We need to 
+enhance this logic to contain connections to each vCenter.  For example, we
+perform checks against each VM to verify hardware version.  Before, all VMs
+would be in the same vCenter.  Now, each node we have, we have to check for the
+VMs existence in the correct vCenter to prevent false negatives.
+
+To solve the multi vCenter check dilemma, we are enhancing each check to make sure
+we take into account this.  We'll create a connection to each vCenter now and
+share these across all the checks.  These connections will be stored in the 
+CheckContext and can be used by any check.
+
+#### vSphere CSI Driver
+
+For complete support of multiple vCenters, the vSphere CSI driver needs to be
+updated to v3.2 in order to get the enhancements made upstream for multiple
+vCenter support.  With the current version of the driver (3.1.x), we will
+have a log message stating that multiple vCenters are not supported yet.
+
+There is already a card for updating the version of the driver to the latest
+version in the backlog of the cluster storage team.  These changes will be 
+tracked separately of this enhancement.
+
+#### vSphere Problem Detector
+
+The vSphere Problem Detector (VPD) will also be updated to handle multiple vCenters
+similar to the vSphere CSI Driver Operator.
+- Enhance to support new YAML cloud provider config
+- Update checks to access multiple vCenters
+- Create new checker to verify infrastructure config and cloud provider config.
+
+As you can see, the changes for VPD follow what was done for vSphere CSI Driver 
+Operator. We are enhancing the operator to use a wrapper config to be able to support
+loading INI and YAML based configs.  This is important for backwards compatability.
+
+All checks have been updated to allow for verifying what vCenter needs to be accessed
+for the various checks.  This can be a bit confusing for the operator as the OCP
+administrator configures the cluster to have a second vCenter.  For clean installs
+with multiple vCenters, the config will be correct out-of-the-box; however, for
+when Day 2 option is persued, the administrator is more likely to make mistake
+updating the cloud provider config or the infrastructure config.  
+
+This leads us to the new checker being added to help the administrator detect when
+such a config mistake may have occurred.  The new check will validate the configs
+in infrastructure and the cloud config to make sure both are configured for the
+same vCenters.  It also makes sure that all failure domains are referencing valid
+vCenters that should be defined in the vCenter section of the infrastructure cluster
+resource.
+
+### Multiple vCenters Configured as Day 2
+
+NOTE: This section is placeholder for future design / work.
```

Comment:
What do we actually support In OCP 4.17? It is possible to edit `Infrastructure` after installation and all the operators try to respect it. But is it supported? Should we block such edits until we figure out what we actually support?